### PR TITLE
Further simplification enabled by chained conversions

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Some simplification enabled by the refactor in #11671. Inspired by review from @JaroslavTulach.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.
- [ ] If meaningful changes were made to logic or tests affecting Enso Cloud integration in the libraries, 
      or the Snowflake database integration, a run of the [Extra Tests](https://github.com/enso-org/enso/actions/workflows/extra-nightly-tests.yml) has been scheduled.
  - If applicable, it is suggested to paste a link to a successful run of the Extra Tests.

### CODE REVIEW

File: lib/rust/parser/src/syntax/token/operator.rs [hunk 1]
```diff
@@ -284,34 +281,13 @@ pub enum Precedence {
     Negation,
     Application,
     Annotation,
-    /// A value that is higher than the precedence of any operator.
-    Max,
+    // NOTE: The highest value must not exceed 0x7e--see usage of `into_u8`.
```
Comment (farmaazon) [inline lib/rust/parser/src/syntax/token/operator.rs:284]: I see only one usage of into_u8: in `ModifiedPrecedence::new`; there's no docs, and I don't know why 0x7f is not allowed.

**Review by farmaazon — APPROVED** [review]

**Review by JaroslavTulach — APPROVED** [review]
Have I provided a useful review for Rust code!?

**Comment by farmaazon** [comment]
> Have I provided a useful review for Rust code!?

I assume what you want to say is that you don't need to be owner of rust parser code.

You were added long ago https://github.com/enso-org/enso/commit/c525b201b9cdc7a5b646e4701ae10e769de0e6f2 by @kazcw ; I guess you may freely unsubscribe from it.

File: lib/rust/parser/src/syntax/token/operator.rs (hunk 1 above)
Comment (kazcw) [reply lib/rust/parser/src/syntax/token/operator.rs:284]: It's unlikely we would ever want to use high numeric values for these, but out of an abundance of caution I put a note here to make sure that anyone considering doing so knows what that could affect. If they are really determined to make that sort of change, the reference to `into_u8` points them in the right direction to find out why they can't use 0x7f.

**Comment by kazcw** [comment]
> Have I provided a useful review for Rust code!?

😆 This isn't the first time you've made some good points--at this point you're a *bona fide* Rustacean. As a Rust developer, when you have to work in Java do you miss the *zero-cost abstractions*?

**Pull request merged**